flink-core/src/test/java/org/apache/flink/api/java/typeutils/runtime/PojoSerializerTest.java

```diff
@@ -50,7 +50,6 @@
 	@Override
 	protected TypeSerializer<TestUserClass> createSerializer() {
 		TypeSerializer<TestUserClass> serializer = type.createSerializer(new ExecutionConfig());
-		assert(serializer instanceof PojoSerializer);
```

Comment:
If I understand correctly, this test originally tested `PojoSerializer`, and now it is testing the generated serializers only. I think it would be good to tweak it to also test PojoSerializer. You could probably do this by creating a subclass of this class, where you override `createSerializer`, and pass an `ExecutionConfig` with codegen disabled.

Additionally, maybe you could also add a test that checks compatibility between `PojoSerializer` and the generated serializers.